library/core/class.dispatcher.php

```
@@ -707,6 +707,19 @@ public function getBlockExceptions() {
         return $this->blockExceptions;
     }
 
+    /**
+     * Adds a dispatcher block exception.
+     *
+     * @param string $exceptionMatch Exception string to match.
+     * @param int $exceptionType Type of block exception.
+     */
+    public function addBlockException(string $exceptionMatch, int $exceptionType) {
+        $allowedExceptionTypes = [self::BLOCK_NEVER, self::BLOCK_PERMISSION, self::BLOCK_ANY];
+        if (in_array($exceptionType, $allowedExceptionTypes)) {
+            $this->blockExceptions[$exceptionMatch] = $exceptionType;
+        }
```

Throw an exception if the type is wrong. Don't silently fail.